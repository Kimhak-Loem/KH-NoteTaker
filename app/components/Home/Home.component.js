import React, {Component} from 'react';
import {View, Text, TextInput, KeyboardAvoidingView} from 'react-native';
import styles from './Home.component.style';
import Notes from '../Notes/Notes.component';
import PropTypes from 'prop-types';
import Touchable from 'react-native-platform-touchable';
import t from '../../utils/language.utils';

class Home extends Component {
  addNote = () => {
    const {saveNote, title, text} = this.props;
    saveNote({title, text});
  };
  render() {
    const {setTitle, title, text, setText, notes, onAboutPress} = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.titleHeading}>{t('HOME_noteTitle')} {t('HOME_greeting', {name: 'John Doe'})}</Text>
        <TextInput
          style={styles.titleTextInput}
          onChangeText={setTitle}
          value={title}
        />
        <Text style={styles.textAreaTitle}> {t('HOME_pleaseTypeYourNote')} </Text>
        <TextInput
          style={styles.textArea}
          multiline={true}
          onChangeText={setText}
          value={text}
        />
        <KeyboardAvoidingView style={styles.bottomBar}>
          <View style={styles.bottomBarWrapper}>
            <Text style={styles.saveBtn} onPress={this.addNote}>
              {t('HOME_save')}
            </Text>
            <Text style={styles.characterCount}>
              {text ? text.length : 0} {t('HOME_characters')}
            </Text>
          </View>
        </KeyboardAvoidingView>
        <Notes data={notes} />
        <Touchable style={styles.aboutUsWrapper} onPress={onAboutPress}>
          <Text style={styles.aboutUs}>{t('HOME_aboutUs')}</Text>
        </Touchable>
      </View>
    );
  }
}

Home.propTypes = {
  setTitle: PropTypes.func,
  saveNote: PropTypes.func,
  setText: PropTypes.func,
  title: PropTypes.string,
  notes: PropTypes.array,
  text: PropTypes.string,
  onAboutPress: PropTypes.func,
};

export default Home;
