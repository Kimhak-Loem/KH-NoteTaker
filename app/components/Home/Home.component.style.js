import { StyleSheet } from 'react-native';

export default StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        padding: 20,
        justifyContent: 'center',
        alignItems: 'center',
    }
});