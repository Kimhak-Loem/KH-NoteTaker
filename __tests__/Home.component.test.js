import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../app/components/Home/Home.component';

describe('Home Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
