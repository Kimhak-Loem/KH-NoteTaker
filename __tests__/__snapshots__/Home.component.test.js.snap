// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text>
     Please enter your note here
  </Text>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 100,
          "width": 200,
        },
        Object {},
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
